enhancements/network/bgp-overview.md

```diff
@@ -0,0 +1,257 @@
+---
+title: bgp-overview
+authors:
+  - "@russellb"
+reviewers:
+  - "@danwinship"
+  - "@squeed"
+approvers:
+  - TBD
+creation-date: 2020-08-11
+last-updated: 2020-08-11
+status: informational
+see-also:
+  - https://github.com/openshift/enhancements/pull/356
+  - https://github.com/openshift/enhancements/pull/394
+---
+
+# OpenShift and BGP
+
+BGP (Border Gateway Protocol) is a dynamic routing protocol used to power a
+number of important networking use cases, including routing in the global
+internet.  For some helpful resources on learning more about BGP, see the
+[BGP Resources](#bgp-resources) section of this document.
+
+The purpose of this document is to serve as a high level overview of how BGP
+relates to OpenShift.  We will review a number of use cases, discuss how they
+could be supported, and link to existing related work.  Any future enhancements
+related to BGP should be able to rely on this one to provide some broader
+context and should continue to update this document as necessary.
+
+## Use Cases
+
+### L3 (Layer 3) Network Fabric
+
+The use case here does not require any changes to OpenShift.  It is still
+useful to call out as it is one use of BGP that is relevant to OpenShift,
+particularly with on premise clusters.
+
+It is possible to use BGP between the routers used to build the L3 network that
+OpenShift Nodes are connected to.
+
+### L3 Redundancy for Nodes
+
+*First, an aside on terminology: This is referring to IP Routing, and
+not the OpenShift feature called Routing that works at the HTTP layer.  There
+are also OpenShift features that use the word "egress", but in this context, we
+only mean to refer to traffic that is leaving the cluster and not related to
+any other features that use that name.*
+
+Some networks may have more than one router accessible from Nodes for
+availability reasons.  In this case an administrator may want to use BGP down
```

Comment:
should we specifically change this BGP call to EBGP, as it'd be outside traffic?

Reply:
I'm not sure it has to be EBGP.  An organization may consider the cluster part of the same AS as the infrastructure it's peering with?

and I think the only real distinction for this use case is how ASNs are configured?